targets/curie_bsp/jerry_app/quark/main.c

```diff
@@ -0,0 +1,168 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+
+/* Infra */
+#include "infra/bsp.h"
+#include "infra/reboot.h"
+#include "infra/log.h"
+#include "infra/time.h"
+#include "infra/system_events.h"
+#include "infra/tcmd/handler.h"
+
+#include "cfw/cfw.h"
+/* Watchdog helper */
+#include "infra/wdt_helper.h"
+
+#include "jerry-api.h"
+#include "string.h"
+#include "jerry-port.h"
+
+#include "zephyr.h"
+#include "microkernel/task.h"
+#include "os/os.h"
+
+static T_QUEUE queue;
+
+jerry_value_t print_function;
+
+void jerry_resolve_error (jerry_value_t ret_value)
+{
+  if (jerry_value_has_error_flag (ret_value))
+  {
+    jerry_value_clear_error_flag (&ret_value);
+    jerry_value_t err_str_val = jerry_value_to_string (ret_value);
+    jerry_char_t err_str_buf[256];
+    jerry_size_t err_str_size = jerry_get_string_size (err_str_val);
+    jerry_size_t sz = jerry_string_to_char_buffer (err_str_val, err_str_buf, err_str_size);
+    err_str_buf[sz] = 0;
+    jerry_port_console ("Script Error: unhandled exception: %s\n", err_str_buf);
+    jerry_release_value (err_str_val);
+  }
+}
+
+
+void help()
+{
+  jerry_port_console ("Usage:\n");
+  jerry_port_console ("js e 'JavaScript Command'\n");
+  jerry_port_console ("eg. js e print ('Hello World');\n");
+}
+
+void eval_jerry_script (int argc, char *argv[], struct tcmd_handler_ctx *ctx)
+{
+  if (argc < 3)
+  {
+    TCMD_RSP_ERROR (ctx, NULL);
+    help ();
+    return;
+  }
+  else
+  {
+    OS_ERR_TYPE err;
+    size_t str_total_length = 0;
+    size_t *str_lens = (size_t *)balloc((argc - 2) * sizeof(size_t), &err);
+    if (str_lens == NULL || err != E_OS_OK)
+    {
+      jerry_port_console ("%s: allocate memory failed!", __func__);
+      TCMD_RSP_ERROR (ctx, NULL);
+      return;
+    }
+    for (int i = 2; i < argc; ++i)
+    {
+      str_lens[i - 2] = strlen(argv[i]);
+      str_total_length += str_lens[i - 2] + 1;
+    }
+    err = E_OS_OK;
+    char *buffer = (char *)balloc(str_total_length, &err);
+    if (buffer == NULL || err != E_OS_OK)
+    {
+      jerry_port_console("%s: allocate memory failed!", __func__);
+      TCMD_RSP_ERROR(ctx, NULL);
+      return;
+    }
+
+    char *p = buffer;
+    for (int i = 2; i < argc; ++i)
+    {
+      for (int j =0; j < str_lens[i - 2]; ++j)
+      {
+        *p = argv[i][j];
+        ++p;
+      }
+      *p = ' ';
+      ++p;
+    }
+    *p = '\0';
+
+    jerry_value_t eval_ret = jerry_eval (buffer, str_total_length - 1, false);
+    jerry_value_t args[] = {eval_ret};
+    jerry_value_t ret_val_print = jerry_call_function (print_function,
```

Comment:
If `eval_ret` is an error, then the result of `jerry_call_function` will be a `TypeError`. I'd move this into the false case of `if (jerry_value_has_error_flag (eval_ret))` at line 125.